### PR TITLE
rnn 10.13 proofread

### PR DESCRIPTION
**Pull request by goldmermaid**

### CODE REVIEW

File: chapter_recurrent-neural-networks/encoder-decoder.md [hunk 1]
```diff
@@ -1,20 +1,19 @@
...
 In this section, we will show an interface to implement this encoder-decoder architecture.
 
-```{.python .input  n=1}
-from mxnet.gluon import nn
-```
 
 ## Encoder
 
-The encoder is a normal neural network that takes inputs, e.g., a source sentence, to return outputs.
+The encoder is a normal neural network that takes inputs, e.g., a source sentence, and then returns states, which is a feature map in a form of vector or tensor.
```
Comment (astonzhang) [inline chapter_recurrent-neural-networks/encoder-decoder.md]: "states" is not a singular, why using "which is"?

File: chapter_recurrent-neural-networks/encoder-decoder.md [hunk 2]
```diff
@@ -27,7 +26,7 @@ class Encoder(nn.Block):
 
 ## Decoder
 
-The decoder has an additional method `init_state` to parse the outputs of the encoder with possible additional information, e.g., the valid lengths of inputs, to return the state it needs. In the forward method, the decoder takes both inputs, e.g., a target sentence, and the state. It returns outputs, with potentially modified state if the encoder contains RNN layers.
+The decoder has an additional method `init_state` to parse the outputs of the encoder (the states) with possible additional information, e.g., returns the state it needs given the valid lengths of inputs. In the forward method, the decoder takes both inputs, e.g., a target sentence and the state. It returns outputs, with potentially modified state if the encoder contains RNN layers.
```
Comment (astonzhang) [inline chapter_recurrent-neural-networks/encoder-decoder.md]: possible additional information, e.g., the valid lengths of inputs

File: chapter_recurrent-neural-networks/encoder-decoder.md [hunk 3]
```diff
@@ -65,8 +64,9 @@ class EncoderDecoder(nn.Block):
 ## Summary
 
 * An encoder-decoder architecture is a neural network design pattern mainly in natural language processing.
-* An encoder is a network (FC, CNN, RNN, etc) that takes the input, and output a feature map, a vector or a tensor.
-* An decoder is a network (usually the same network structure as encoder) that takes the feature vector from the encoder, and gives the best closest match to the actual input or intended output.
+* An encoder is a neural network (FC, CNN, RNN, etc.) that takes the input, and output a feature map.
+* An decoder is a network usually the same network structure as encoder.
```
Comment (astonzhang) [inline chapter_recurrent-neural-networks/encoder-decoder.md]: An decoder is a network usually the same network structure as encoder. 

this is not a sentence

File: chapter_recurrent-neural-networks/encoder-decoder.md (hunk 1 above)
Comment (goldmermaid) [reply chapter_recurrent-neural-networks/encoder-decoder.md]: noticed

File: chapter_recurrent-neural-networks/encoder-decoder.md (hunk 2 above)
Comment (goldmermaid) [reply chapter_recurrent-neural-networks/encoder-decoder.md]: sure

File: chapter_recurrent-neural-networks/encoder-decoder.md (hunk 3 above)
Comment (goldmermaid) [reply chapter_recurrent-neural-networks/encoder-decoder.md]: ok

**Pull request merged**